rdr_service/genomic/genomic_job_controller.py

```diff
@@ -476,140 +479,219 @@ def _set_report_type(records):
                 logging.warning(f'Cannot set report state type: message record{records[0].messageRecordId}: {e}')
                 return report_state
 
-        if 'informing_loop' in event_type:
+        if self.job_id == GenomicJob.INGEST_INFORMING_LOOP:
             loop_type = event_type
-            informing_records = self.message_broker_event_dao.get_informing_loop(
+            informing_loop_records = self.message_broker_event_dao.get_informing_loop(
                 message_record_id,
                 loop_type
             )
 
-            if informing_records:
-                module_type = _set_module_type(informing_records)
-                if not module_type:
-                    logging.warning(f'Cannot find module type in message record id: '
-                                    f'{informing_records[0].messageRecordId}')
-                    self.job_result = GenomicSubProcessResult.ERROR
-                    return
+            if not informing_loop_records:
```

Comment:
Can you add a slack alert to these conditionals (for IL, Results, and Appointments) when a record matching the `messageRecordId` isn't found? We have some discrepancies between the IL table and the Reconciliation process and I am wondering if there is a race condition happening where data retrieval is attempted before replication completes.

Reply:
Sounds good.
https://github.com/all-of-us/raw-data-repository/pull/3090/commits/d7e072fe1c0276a80e92667ebc57d7231dcecfc9

Reply:
Thanks, this will help a lot in researching the reconciliation discrepancies going forward.